### PR TITLE
Added documentation for immediate feedback

### PR DESCRIPTION
**Pull request by bsekachev**
### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Introduced immediate feedback for annotators, providing performance scores after job completion.
	- Added the ability for annotators to re-annotate jobs based on feedback.
	- Configurable settings for feedback quality on the quality settings page.

- **Documentation**
	- New documentation created for immediate feedback, outlining usage, settings, and requirements.

### CODE REVIEW

File: site/content/en/docs/enterprise/immediate-feedback.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: 'Immediate job feedback'
+linkTitle: 'Immediate job feedback'
+weight: 5
+description: 'This feature provides annotators with general feedback on their performance in a job.'
+---
+
+When an annotator finishes a job, a dialog is displayed showing the quality of their annotations.
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply site/content/en/docs/enterprise/immediate-feedback.md:8]: That is not true. You may see the dialog when click "Finish the job" in any mode.

File: site/content/en/docs/enterprise/immediate-feedback.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+- Finally rejected when the number of attempts is exhausted
+
+<img src="/images/immediate-feedback-accept.png" style="max-width: 300px;">
+<img src="/images/immediate-feedback-reject.png" style="max-width: 300px;">
+<img src="/images/immediate-feedback-final-reject.png" style="max-width: 300px;">
+
+Notes:
+
+> Immediate feedback has a default timeout of 20 seconds.
+Feedback may be unavailable for large jobs or when there are too many immediate feedback requests.
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply site/content/en/docs/enterprise/immediate-feedback.md:34]: > what will happen if feedback is unavailable

Literally, nothing. If it is unavailable, user will not see it. 
The job is not accepted and not rejected. 

In any scenario the state changes to `completed`.

File: site/content/en/docs/enterprise/immediate-feedback.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+title: 'Immediate job feedback'
+linkTitle: 'Immediate job feedback'
+weight: 5
+description: 'This feature provides annotators with general feedback on their performance in a job.'
+---
+
+When an annotator finishes a job, a dialog is displayed showing the quality of their annotations.
+The annotator can either agree or disagree with the feedback.
+If they disagree, they have the option to re-annotate the job and request feedback again.
+However, feedback is only available a limited number of times, as specified in the task's quality settings.
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply site/content/en/docs/enterprise/immediate-feedback.md:11]: I would prefer to update it first and move to paid features also as it is not available on old place anymore

File: site/content/en/docs/enterprise/immediate-feedback.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (klakhov) [reply site/content/en/docs/enterprise/immediate-feedback.md:8]: Ok, I see

File: site/content/en/docs/enterprise/immediate-feedback.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (klakhov) [reply site/content/en/docs/enterprise/immediate-feedback.md:11]: Then we can add a comment here so we dont forget to add it later

File: site/content/en/docs/enterprise/immediate-feedback.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (klakhov) [reply site/content/en/docs/enterprise/immediate-feedback.md:34]: Ah, I've mixed up `annotations submited` modal with `immediate feedback` modal in single shape mode.

Anyway, we can add here that no feedback will be shown in such case.

**Pull request merged**